File: golang/create/prototype.go
package main
type ProtoType struct {
    id int
    name string
}
func (p *ProtoType) Clone() ProtoType {
    return ProtoType{p.id, p.name}
}
func main() {
    p := ProtoType{1, "hello"}
    c := p.Clone()
    println(c.id, c.name)
}

File: cpp/create/prototype.cpp
#include<iostream>
using namespace std;

class ProtoType {
private:
    int id;
    string name;
public:
    ProtoType(int _id, string _name):id(_id), name(_name) {}
    int getId() {return id;}
    string getName() {return name;}
    ProtoType& operator=(const ProtoType& p) {
        id = p.id;
        name = p.name;
        return *this;
    }
};
int main() {
    ProtoType p(1, "hello");
    ProtoType c = p;
    cout<<c.getId() <<":" <<c.getName() <<endl;

}